install/azure.go

```diff
@@ -110,6 +110,18 @@ func (k *K8sInstaller) createAzureServicePrincipal(ctx context.Context) error {
 			k.params.Azure.ClientID, k.params.Azure.TenantID)
 	}
 
+	/*
+		`az aks create` requires an existing resource group in which to create a
+		new AKS cluster, but a single resource group may hold multiple AKS clusters.
+
+		Internally, AKS creates an intermediate resource group (named
+		`MC_{RG_name}_{cluster_name}_{location}`) to regroup all AKS nodes for
+		this cluster.
+
+		The CLI installs itself into this intermediate resource group, and thus
+		derives it from the user-given resource group and cluster name using
+		`az aks show`.
+	*/
```

Comment:
Note we don't generally use that format for comments. Seems to be the first use in both cilium-cli and cilium/cilium.

Reply:
OK, what format to we use?

Reply:
The C-style formats it seems :shrug: 

It seems to be `//` in front of each line most of the time

Reply:
Aaaah OK I thought you were talking about something else, my bad 🤦🏻 I will switch to `//` in next push, just waiting for Timo's review in case there are other nits.

Reply:
I even forgot Go supported this, hadn't seen it in so long. :grin: